src/graphblas/reference/CMakeLists.txt

```diff
@@ -22,84 +22,13 @@
 # in its compilation interface.
 #
 
-assert_valid_variables( BACKEND_LIBRARY_OUTPUT_NAME VERSION
-	SHMEM_BACKEND_INSTALL_DIR INCLUDE_INSTALL_DIR
+assert_valid_variables( SHMEM_BACKEND_INSTALL_DIR
 	REFERENCE_BACKEND_DEFAULT_NAME REFERENCE_OMP_BACKEND_DEFAULT_NAME
-	REFERENCE_SELECTION_DEFS REFERENCE_OMP_SELECTION_DEFS backend_reference_srcs
+	REFERENCE_SELECTION_DEFS REFERENCE_OMP_SELECTION_DEFS
 )
 
 assert_defined_targets( backend_flags )
 
-# macro to create a target for the shared memory library, with all minimal properties
-#
-# Parameters:
-#   targetName name of the target
-#   targetType type of library (STATIC or SHARED)
-#   outDir directory to compile the binary to (not install)
-#
-macro( make_reference_target targetName targetType outDir )
-
-	add_library( "${targetName}" "${targetType}"
-		"${backend_reference_srcs}"
-	)
-	set_target_properties( "${targetName}" PROPERTIES
-		OUTPUT_NAME "${BACKEND_LIBRARY_OUTPUT_NAME}"
-	)
-	if( "${targetType}" STREQUAL "SHARED" )
-		set_target_properties( "${targetName}" PROPERTIES
-			SOVERSION "${VERSION}"
-			LIBRARY_OUTPUT_DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}/${outDir}"
-		)
-	elseif( "${targetType}" STREQUAL "STATIC" )
-		set_target_properties( "${targetName}" PROPERTIES
-			ARCHIVE_OUTPUT_DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}/${outDir}"
-		)
-	else()
-		message( FATAL_ERROR "Unknown library type: ${targetType}")
-	endif()
-	target_link_libraries( "${targetName}" PRIVATE backend_flags )
-
-endmacro( make_reference_target )
-
-
-### BINARY-ONLY TARGETS: create the actual library with all code inside
-### but storing only the minimum interface definitions (e.g. no OMP);
-### no default backend is set!
-
-if( ${CMAKE_VERSION} VERSION_GREATER_EQUAL "3.19.0" )
-
-	# alias target for basic propagation of headers and definitions
-	# from existing header target, which depends on the enabled backends;
-	# use the most advanced ones, i.e. those of reference_omp if available
-	if( WITH_OMP_BACKEND_HEADERS )
-		set( backend_shmem_base_headers backend_reference_omp_headers)
-	elseif( WITH_REFERENCE_BACKEND_HEADERS )
-		set( backend_shmem_base_headers backend_reference_headers )
-	endif()
-
-	## STATIC
-	make_reference_target( backend_shmem_static STATIC "shmem" )
-	target_link_libraries( backend_shmem_static PRIVATE ${backend_shmem_base_headers} )
-	# this is the actual binary file, i.e. the one to be installed
-	install( TARGETS backend_shmem_static
-		EXPORT GraphBLASTargets
-		ARCHIVE DESTINATION "${SHMEM_BACKEND_INSTALL_DIR}"
-	)
-
-	## DYNAMIC
-	make_reference_target( backend_shmem_shared SHARED "shmem" )
-	target_link_libraries( backend_shmem_shared PRIVATE ${backend_shmem_base_headers} )
-
-	install( TARGETS backend_shmem_shared
-		EXPORT GraphBLASTargets
-		LIBRARY DESTINATION "${SHMEM_BACKEND_INSTALL_DIR}"
-	)
-
-	add_dependencies( libs backend_shmem_static )
-	add_dependencies( libs backend_shmem_shared )
-endif()
```

Comment:
See also the above comment -- should this part actually not stay in this file, with only the macro moving up to `src/graphblas/CMakeLists.txt` ?